addons/binding/org.openhab.binding.tadoac/src/main/java/org/openhab/binding/tadoac/handler/TadoACHandler.java

```diff
@@ -0,0 +1,216 @@
+                    break;
+                default:
+                    logger.error("mode channel out of range");
+            }
+        }
+
+        synchronized (this) {
+            if (sendFuture != null) {
+                sendFuture.cancel(false);
+            }
+            sendFuture = scheduler.schedule(() -> sendSetting(), 1, TimeUnit.SECONDS);
+        }
+    }
+
+    private synchronized void sendSetting() {
+        try {
+            String username = (String) getConfig().get("username");
+            String password = (String) getConfig().get("password");
+            int homeid = ((BigDecimal) getConfig().get("homeid")).intValue();
+            int zoneid = ((BigDecimal) getConfig().get("zoneid")).intValue();
+            if (username == null || password == null) {
+                logger.error("login credentials not configured");
+                updateStatus(ThingStatus.OFFLINE);
+                return;
+            }
+
+            // validate setting
+            if (setting.getPower() == null) {
+                logger.error("No power setting");
+                return;
+
+            }
+            if (setting.getPower() == TadoACPower.ON) {
+                if (setting.getMode() == null) {
+                    logger.error("No mode setting");
+                    return;
+                }
+                if (setting.getFanSpeed() == null) {
+                    logger.error("No fan setting");
+                    return;
+                }
+                if (setting.getMode() == TadoACMode.COOL && setting.getTemperature() == null) {
+                    logger.error("No temperature setting");
+                    return;
+                }
+            }
+            TadoACConnector connector = new TadoACConnector(username, password);
+            connector.setSetting(homeid, zoneid, setting);
+            updateStatus(ThingStatus.ONLINE);
+        } catch (Exception e) {
+            e.printStackTrace();
+            logger.error("could not send state", e);
+        } finally {
+            sendFuture = null;
+        }
+
+    }
+
+    private synchronized void refreshSetting() {
+        try {
+            String username = (String) getConfig().get("username");
+            String password = (String) getConfig().get("password");
+            int homeid = ((BigDecimal) getConfig().get("homeid")).intValue();
+            int zoneid = ((BigDecimal) getConfig().get("zoneid")).intValue();
+            if (username == null || password == null) {
+                logger.error("login credentials not configured");
+                updateStatus(ThingStatus.OFFLINE);
+                return;
+            }
+            TadoACConnector connector = new TadoACConnector(username, password);
+            TadoACSetting newSetting = connector.getSetting(homeid, zoneid);
+
+            // Check each part and copy them to the previous state (No loss of data)
+            if (newSetting.getPower() != null) {
```

Comment:
Please store the result of `newSetting.getPower()` in a local variable if you use it more than once